backup-stores/s3/src/test/java/io/camunda/zeebe/backup/s3/BackupUploadIT.java

```diff
@@ -0,0 +1,171 @@
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.backup.s3;
+
+import io.camunda.zeebe.backup.api.Backup;
+import io.camunda.zeebe.backup.common.BackupDescriptorImpl;
+import io.camunda.zeebe.backup.common.BackupIdentifierImpl;
+import io.camunda.zeebe.backup.common.BackupImpl;
+import io.camunda.zeebe.backup.common.NamedFileSetImpl;
+import io.camunda.zeebe.backup.s3.S3BackupConfig.Builder;
+import io.camunda.zeebe.test.util.testcontainers.ContainerLogsDumper;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.apache.commons.lang3.RandomUtils;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.testcontainers.containers.localstack.LocalStackContainer;
+import org.testcontainers.containers.localstack.LocalStackContainer.Service;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.utility.DockerImageName;
+import software.amazon.awssdk.services.s3.S3AsyncClient;
+import software.amazon.awssdk.services.s3.model.CreateBucketRequest;
+
+@Testcontainers
+final class BackupUploadIT {
+  private static final String BUCKET_NAME = RandomStringUtils.randomAlphabetic(10).toLowerCase();
+
+  @Container
+  private static final LocalStackContainer S3 =
+      new LocalStackContainer(DockerImageName.parse("localstack/localstack"))
+          .withServices(Service.S3)
+          .withEnv("LS_LOG", "trace");
+
+  @RegisterExtension
+  final ContainerLogsDumper logsWatcher = new ContainerLogsDumper(() -> Map.of("localstack", S3));
+
+  private S3AsyncClient client;
+  private S3BackupStore store;
+  private S3BackupConfig config;
+
+  public S3AsyncClient getClient() {
+    return client;
+  }
+
+  public S3BackupConfig getConfig() {
+    return config;
+  }
+
+  public S3BackupStore getStore() {
+    return store;
+  }
+
+  @BeforeAll
+  static void setupBucket() {
+    final var config =
+        new Builder()
+            .withBucketName(BUCKET_NAME)
+            .withEndpoint(S3.getEndpointOverride(Service.S3).toString())
+            .withRegion(S3.getRegion())
+            .withCredentials(S3.getAccessKey(), S3.getSecretKey())
+            .build();
+    try (final var client = S3BackupStore.buildClient(config)) {
+      client.createBucket(CreateBucketRequest.builder().bucket(config.bucketName()).build()).join();
+    }
+  }
+
+  @AfterEach
+  void tearDown() {
+    store.closeAsync();
+  }
+
+  public void setConfigParallelConnectionsAndTimeout(
```

Comment:
:wrench: Instead of setting the `store` field, just return the `store` and rename to `buildStore` or something similar.

Reply:
We still use the store for the teardown(), I left it just to have a similar implementation to other tests, but it would also work as you suggested since it is within the scope of the test. Let me know if I should remove it or leave it as is.

Reply:
:+1: Your call - I prefer the explicit handling in each test but both are okay.
If you keep the current style, please rename `setConfigParallelConnectionsAndTimeout` anyway. This method does not just set two config parameters, it builds the entire backup store.